### PR TITLE
Nautobot 2 Migration

### PR DESCRIPTION
**Pull request by snaselj**
# Closes NaN

# What's Changed

Updated, to be used with Nautobot 2.0.

# To Do

- [ ] Fix `TBD`s in the code.
- [x] Update the documentation.
- [x] Improve change fragments.
- [ ] Update CI Nautobot and Python versions.
- [ ] Test manually.

### CODE REVIEW

**Comment by snaselj** [comment]
This PR has some To-Do's yet to be done, however it's possible to start reviewing it now.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -82,8 +82,8 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        python-version: ["3.8"]
-        nautobot-version: ["1.4.0"]
+        python-version: ["3.10"]
```
Comment (chadell) [inline .github/workflows/ci.yml:85]: why 3.10?

File: nautobot_circuit_maintenance/__init__.py [hunk 2]
```diff
@@ -35,7 +34,9 @@ def custom_fields_extension(sender, *, apps=global_apps, **kwargs):  # pylint: d
             "label": "Provider Parser for Circuit Maintenance plugin.",
         },
     ]:
-        field, _ = CustomField.objects.get_or_create(name=provider_cf_dict["name"], defaults=provider_cf_dict)
+        defaults = {**provider_cf_dict}
+        name = defaults.pop("name")
+        field, _ = CustomField.objects.get_or_create(label=name, defaults=defaults)
```
Comment (chadell) [inline nautobot_circuit_maintenance/__init__.py:39]: what does `label` do?

File: nautobot_circuit_maintenance/api/urls.py [hunk 3]
```diff
@@ -8,6 +8,8 @@
 router.register("note", views.MaintenanceNoteTaskView)
 router.register("circuitimpact", views.MaintenanceCircuitImpactTaskView)
 router.register("notificationsource", views.NotificationSourceTaskView)
+router.register("parsednotification", views.ParsedNotificationTaskView)
```
Comment (chadell) [inline nautobot_circuit_maintenance/api/urls.py:11]: why do we need to add these?

File: nautobot_circuit_maintenance/tests/utils.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    """Mocked logger for testing."""
+
+    def __init__(self):
+        self.debug = _add("debug")
+        self.info = _add("info")
+        self.warning = _add("warning")
+        self.error = _add("error")
+
+
+class MockedJob:
```
Comment (chadell) [inline nautobot_circuit_maintenance/tests/utils.py:24]: nice :)
This could come directly from Nautobot to support testing

File: pyproject.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "nautobot-circuit-maintenance"
-version = "1.0.0"
+version = "2.0.0b1"
```
Comment (chadell) [inline pyproject.toml:3]: why not b2?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (snaselj) [reply .github/workflows/ci.yml:85]: I'm testing with this version on my local, updates to this are mentioned in `To Do`s in this PR.

File: nautobot_circuit_maintenance/api/urls.py (hunk 3 above)
Comment (snaselj) [reply nautobot_circuit_maintenance/api/urls.py:11]: Some tests were failing, trying to access these URLs, so I added it here.

File: nautobot_circuit_maintenance/__init__.py (hunk 2 above)
Comment (snaselj) [reply nautobot_circuit_maintenance/__init__.py:39]: `name` has been removed from `CustomField`, and `key` is populated from `label`.

Documentation here should be updated: https://github.com/nautobot/nautobot/blob/next/nautobot/docs/development/apps/migration/model-updates/extras.md#update-customfield-computedfield-and-relationship

File: nautobot_circuit_maintenance/tests/utils.py (hunk 4 above)
Comment (snaselj) [reply nautobot_circuit_maintenance/tests/utils.py:24]: Thanks, with @bryanculver we agreed not to do it now.

File: development/nautobot_config.py [hunk 6]
```diff
@@ -61,13 +61,20 @@
...
-DEBUG_TOOLBAR_CONFIG = {"SHOW_TOOLBAR_CALLBACK": lambda _request: DEBUG and not TESTING}
-
-if DEBUG and "debug_toolbar" not in INSTALLED_APPS:  # noqa: F405
-    INSTALLED_APPS.append("debug_toolbar")  # noqa: F405
-if DEBUG and "debug_toolbar.middleware.DebugToolbarMiddleware" not in MIDDLEWARE:  # noqa: F405
-    MIDDLEWARE.insert(0, "debug_toolbar.middleware.DebugToolbarMiddleware")  # noqa: F405
+if DEBUG:
+    try:
+        # pylint: disable-next=import-outside-toplevel
+        import better_exceptions
```
Comment (snaselj) [inline development/nautobot_config.py:67]: Should I keep `better-exceptions` here? It helps with debugging, displays arguments passed within stack trace output.

Comment (chadell) [reply development/nautobot_config.py:67]: if it comes from our best developer practices, let's do it :)

Comment (snaselj) [reply development/nautobot_config.py:67]: Will remove it here and open an issue to Nautobot Core

Comment (snaselj) [reply development/nautobot_config.py]: https://github.com/nautobot/nautobot/issues/4296

**Comment by cmsirbu** [comment]
If only testing is pending here, @chadell may I suggest merging and cutting a pre-release so it can be tested as a regular package install?

**Comment by chadell** [comment]
ok, let's do that

**Pull request merged**